pose3D pose
float64[] reference
